src/main/java/org/mycore/externalstore/model/MCRExternalStoreFileInfo.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.externalstore.model;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * An {@link MCRExternalStoreFileInfo} describes a file.
+ * Also, a file can be a directory.
+ */
+public class MCRExternalStoreFileInfo {
```

Comment:
Maybe it could be a record.

Reply:
There are a few methods where this becomes slightly hacky as a result, but it's basically a good idea.